opentelemetry-resourcedetector-gcp/tests/test_gke.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from unittest.mock import MagicMock
+
+import pytest
+from opentelemetry.resourcedetector.gcp_resource_detector import _gke
+
+
+# Reset stuff before every test
+# pylint: disable=unused-argument
+@pytest.fixture(autouse=True)
+def autouse(fake_get_metadata, fake_environ):
+    pass
+
+
+def test_detects_on_gke() -> None:
+    os.environ["KUBERNETES_SERVICE_HOST"] = "fakehost"
```

Comment:
Shouldn't we also verify we have access to the metadata server here?

Reply:
I think the discussion above applies here